pkg/controllers/management/clusterstatus/temporarycredentials.go

```diff
@@ -0,0 +1,46 @@
+package clusterstatus
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+)
+
+const temporaryCredentialsAnnotationKey = "clusterstatus.management.cattle.io/temporary-security-credentials"
+
+func Register(management *config.ManagementContext) {
+	c := &clusterAnnotations{
+		clusters: management.Management.Clusters(""),
+	}
+
+	management.Management.Clusters("").AddHandler("temporary-credentials", c.sync)
+}
+
+type clusterAnnotations struct {
+	clusters v3.ClusterInterface
+}
+
+func (cd *clusterAnnotations) sync(key string, cluster *v3.Cluster) error {
+	if key == "" || cluster == nil {
+		return nil
+	}
+
+	original := cluster
+	cluster = original.DeepCopy()
```

Comment:
can we do a deep copy only when we decide to update? would save us on performance